src/Transaction.php

```diff
@@ -5,69 +5,54 @@
 class Transaction {
 
     /**
-     * @var string The domain of the transaction.
+     * @var string The required keys for the transaction.
      */
-    private $domain;
+    private $requiredKeys = [
+        'domain', 
+        'url', 
+        'name', 
+        'articleEID', 
+        'magazineKey', 
+        'checkoutToken'
+    ];
 
     /**
-     * @var string The URL of the transaction.
+     * @var string The payload of the transaction.
      */
-    private $url;
+    private $payload;
 
     /**
-     * @var string The name associated with the transaction.
-     */
-    private $name;
-
-    /**
-     * @var string The external article ID.
+     * NewsLadderTransaction constructor.
+     * 
+     * @param string $payload ['domain', 'url', 'name', 'articleEID', 'magazineKey', 'checkoutToken']
      */
-    private $external_article_id;
+    public function __construct(array $payload) {
+        if (empty($payload)) {
+            throw new \InvalidArgumentException("Payload is empty");
+        }
 
-    /**
-     * @var string The magazine key.
-     */
-    private $magazine_key;
+        if (count($this->requiredKeys) < count($payload)) {
+            throw new \InvalidArgumentException("Too many keys in payload");
+        }
 
-    /**
-     * @var string The checkout token.
-     */
-    private $checkout_token;
+        foreach ($this->requiredKeys as $key) {
+            if (!array_key_exists($key, $payload)) {
```

Comment:
What if the key is available but empty? We should catch that as well, imo...

Reply:
We could Check against a regex pattern.

Reply:
I think a simple `isset($this->requiredKeys[$key] && !empty($this->requiredKeys[$key])` is sufficient for the time being...